client/webserver/site/src/js/app.ts

```diff
@@ -934,12 +925,25 @@ function setSeverityClass (el: HTMLElement, severity: number) {
   el.classList.add(severityClassMap[severity])
 }
 
+// Updates given order in market's orders list if it finds it.
+// Returns a bool which indicates if order was found.
+function updateOrder (mkt: Market, ord: Order): boolean {
+  for (const i in mkt.orders || []) {
+    if (mkt.orders[i].id === ord.id) {
+      mkt.orders[i] = ord
+      return true
+    }
+  }
+  return false
+}
+
 /* updateMatch updates the match in or adds the match to the order. */
-function updateMatch (order: Order, match: Match) {
+function updateMatch (order: Order, match: Match, mkt: Market) {
   for (const i in order.matches) {
     const m = order.matches[i]
     if (m.matchID === match.matchID) {
       order.matches[i] = match
+      updateOrder(mkt, order)
```

Comment:
Hmm. Does this do anything? `updateMatch` is only used once, in the `'match'` notification handler. The order there is from 
```Javascript
const order = this.order(n.orderID)
```
which itself does, 
```Javascript
order (oid: string): Order | null {
    for (const xc of Object.values(this.user.exchanges)) {
      for (const market of Object.values(xc.markets)) {
        if (!market.orders) continue
        for (const ord of market.orders) {
          if (ord.id === oid) return ord
        }
      }
    }
    return null
  }
```
, and the market passed in is similarly from the map
```Javascript
const mkt = this.user.exchanges[order.host].markets[order.market]
```

These are all reference values, so you're simply replacing an order with itself, with no net effect, I think.

The thing is, this line is really the only change here, so if this does nothing, all of the changes in this file do nothing.

Reply:
That's correct. I thought the reason the value wasn't updating could be because it's reference was not updating, but I was wrong.

Thanks for pointing that out.